jazz_codeq/index.js

```diff
@@ -0,0 +1,211 @@
+					return cb(errorMessage);
+				} else {
+					const metrics = result.metrics;
+					const toDate = result.toDate;
+					const fromDate = result.fromDate;
+					const query = result.query;
+					let authToken;
+					let baseAuthToken;
+					return utils.getSonarToken(config)
+					.then((data) => {
+						baseAuthToken = data.base_auth_token;
+						return utils.getEnvironmentToken(config);
+					}).then((data) => {
+						authToken = data.auth_token;
+						return utils.getProjectBranch(authToken, query, config);
+					}).then(data => utils.getCodeqReport(metrics, data.branch, baseAuthToken, toDate, fromDate, query, config)
+					).then(data => {
+						const output = responseObj(data.get_codeq_report, serviceContext.query);
+						return cb(null, output);
+					}).catch(err => {
+						const output = getReportOnError(err, metrics, config, serviceContext);
+						if(output.error) {
+							return cb(JSON.stringify(errorHandler.throwInputValidationError(output.error)));
+						} else {
+							return cb(null, output);
+						}
+					});
+				}
+			//pathstring == 'help'
+			} else if (serviceContext && serviceContext.method && serviceContext.method === 'GET' && pathString === "help") {
+				logger.info("code quality help service called.");
+				const result = getResponseForHelpPathString(serviceContext, config);
+				if (result.error) {
+					return cb(JSON.stringify(errorHandler.throwInputValidationError(result.error)));
+				} else {
+					return cb(null, result);
+				}
+			} else {
+				return cb(JSON.stringify(errorHandler.throwInputValidationError(messages.SERVICE_INPUT_ERROR)));
+			}
+		}).catch(err => {
+			return cb(JSON.stringify(errorHandler.throwInputValidationError(err.errorMessage)));
+		});
+	} catch (e) {
+		return cb(JSON.stringify(errorHandler.throwInternalServerError(e)));
+	}
+};
+
+
+function getResponseForHelpPathString(serviceContext, config) {
+	const query = utils.getQuery(serviceContext);
+	const result = utils.getMetrics(query, config, messages);
+
+	if (result.error) {
+		logger.error(result.error);
+		return result;
+	}
+	
+	const metrics = result.metrics;
+	let output = { metrics: []};
+
+	for (let q = metrics.length - 1; q >= 0; q--) {
+		const metricsHelp = config.METRICS_HELP[metrics[q]];
+		
+		output.metrics.push({
+			"name": metrics[q],
+			"description": metricsHelp.description,
+			"unit": metricsHelp.unit,
+			"minValue": metricsHelp.minValue,
+			"maxValue": metricsHelp.maxValue
+		});
+	}
+
+	return responseObj(output, serviceContext.query);
+}
+
+function getCodeqInputsUsingQuery(serviceContext, config) {
+	let query = utils.getQuery(serviceContext);
+	let result = {};
+	// validate required fields
+	const missingRequiredFields = validation.validateMissingFields(config.REQUIRED_PARAMS, query);
+	if (missingRequiredFields) {
+		let message = messages.MISSING_FIELDS + missingRequiredFields;
+		logger.error(message);

```

Comment:
remove this `logger.error` so that we stay consistent with the rest of the code in this function. Let the caller log the error.